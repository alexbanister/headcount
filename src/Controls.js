import React, { Component } from 'react';
import DatasetButton from './DatasetButton';
import PropTypes from 'prop-types';

class Controls extends Component {
  constructor() {
    super();
    this.state = {
      dataSet: '',
      searchValue: ''
    };
  }

  handleChange(event) {
    this.setState({dataSet: event.target.value});
    this.props.changeData(event.target.value);
  }

  handleSearch (event) {
    this.setState({searchValue: event.target.value});
    this.props.searchDistricts(event.target.value);
  }

  render() {
    const fileListDropdown = this.props.fileList.map( (fileName) => {
      return (
        <DatasetButton
          key={fileName}
          dataSet={fileName} />
      );
    });
    return (
      <div className='header'>
        <h1>Welcome To Headcount 2.0</h1>
        <select
          value={this.state.dataSet}
          onChange={this.handleChange.bind(this)}>

          <option value=''>Select a Dataset</option>
          {fileListDropdown}
        </select>
        <input
          value={this.state.searchValue}
          onChange={this.handleSearch.bind(this)} />
      </div>
    );
  }
}

Controls.propTypes = {
  searchDistricts: React.PropTypes.func,
  fileList: React.PropTypes.arrayOf(PropTypes.string),
  changeData: React.PropTypes.func
};

export default Controls;
