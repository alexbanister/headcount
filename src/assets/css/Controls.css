h1 {
  font-size: 3em; }

.header {
  text-align: center;
  padding: 2em;
  margin-bottom: 2em;
  border-bottom: 2px solid #000000; }
