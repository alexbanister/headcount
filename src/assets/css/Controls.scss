@import "variables.scss";

h1 {
  font-size: 3em;
  text-shadow: 1px 1px 5px $dark-grey;
}

.header {
  @include linear-gradient($dark-grey, $light-grey);
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  justify-content: space-between;
  padding: 2em;
  border-bottom: 2px solid $dark-grey;
  h1 {
    width: 100%;
    text-align: center;
    margin-bottom: 1em;
    strong {
      color: $select-color-bright;
    }
  }
  select {
    @include linear-gradient($highlight-dark, $highlight);
    width: 20em;
    height: 3em;
    font-size: 1em;
    padding: 1em;
    color: $dark-grey;
    border: 1px solid $dark-grey;
  }
  input {
    font-size: 1em;
    width: 20em;
    padding: .5em;
    color: $dark-grey;
  }
}
