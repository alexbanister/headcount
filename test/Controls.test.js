import React from 'react';
import { shallow, mount } from 'enzyme';
import Controls from '../src/Controls';
import DistrictRepository from '../src/helper.js';
import kinderData from '../data/kindergartners_in_full_day_program.js';

describe('Controls', () => {
  const district = new DistrictRepository(kinderData);

  it('should create an instance of a card container', () => {
    const mockFunc = jest.fn();
    const wrapper = shallow(<Controls
      searchDistricts={mockFunc}
      fileList={Object.keys(district)}
      changeData={mockFunc}
    />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should change state if user types into search field', () => {
    const mockFunc = jest.fn();
    const wrapper = mount(<Controls
      searchDistricts={mockFunc}
      fileList={Object.keys(district)}
      changeData={mockFunc}
    />);
    const searchInput = wrapper.find('.search-input');

    expect(wrapper.state().searchValue).toEqual('');
    searchInput.simulate('change', { target: { value: 'aca' } });
    expect(wrapper.state().searchValue).toEqual('aca');
  });
});
