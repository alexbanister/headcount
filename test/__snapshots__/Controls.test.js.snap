exports[`Controls should create an instance of a card container 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Controls
    changeData={[Function]}
    fileList={
        Array [
            "districtData",
          ]
    }
    searchDistricts={[Function]} />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Headcount 
          <strong>
                    2.0
          </strong>
</h1>,
        <select
          onChange={[Function]}
          value="">
          <option
                    value="">
                    Select a Dataset
          </option>
          <DatasetButton
                    dataSet="districtData" />
</select>,
        <input
          className="search-input"
          onChange={[Function]}
          placeholder="Search Districts"
          value="" />,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Headcount ",
            <strong>
              2.0
</strong>,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Headcount ",
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "2.0",
            },
            "ref": null,
            "rendered": "2.0",
            "type": "strong",
          },
        ],
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              value="">
              Select a Dataset
</option>,
            Array [
              <DatasetButton
                dataSet="districtData" />,
            ],
          ],
          "onChange": [Function],
          "value": "",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Select a Dataset",
              "value": "",
            },
            "ref": null,
            "rendered": "Select a Dataset",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "districtData",
            "nodeType": "function",
            "props": Object {
              "dataSet": "districtData",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "select",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "className": "search-input",
          "onChange": [Function],
          "placeholder": "Search Districts",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Headcount 
            <strong>
                        2.0
            </strong>
</h1>,
          <select
            onChange={[Function]}
            value="">
            <option
                        value="">
                        Select a Dataset
            </option>
            <DatasetButton
                        dataSet="districtData" />
</select>,
          <input
            className="search-input"
            onChange={[Function]}
            placeholder="Search Districts"
            value="" />,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Headcount ",
              <strong>
                2.0
</strong>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Headcount ",
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "2.0",
              },
              "ref": null,
              "rendered": "2.0",
              "type": "strong",
            },
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option
                value="">
                Select a Dataset
</option>,
              Array [
                <DatasetButton
                  dataSet="districtData" />,
              ],
            ],
            "onChange": [Function],
            "value": "",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Select a Dataset",
                "value": "",
              },
              "ref": null,
              "rendered": "Select a Dataset",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "districtData",
              "nodeType": "function",
              "props": Object {
                "dataSet": "districtData",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "select",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "className": "search-input",
            "onChange": [Function],
            "placeholder": "Search Districts",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
